pkg/apis/application/v1alpha1/types.go

```diff
@@ -425,8 +425,9 @@ func (k *ApplicationSourceKsonnet) IsZero() bool {
 }
 
 type ApplicationSourceDirectory struct {
-	Recurse bool                     `json:"recurse,omitempty" protobuf:"bytes,1,opt,name=recurse"`
-	Jsonnet ApplicationSourceJsonnet `json:"jsonnet,omitempty" protobuf:"bytes,2,opt,name=jsonnet"`
+	Recurse    bool                     `json:"recurse,omitempty" protobuf:"bytes,1,opt,name=recurse"`
+	Jsonnet    ApplicationSourceJsonnet `json:"jsonnet,omitempty" protobuf:"bytes,2,opt,name=jsonnet"`
+	Exclusions []string                 `json:"exclusions,omitempty" protobuf:"bytes,3,opt,name=exclusions"`
```

Comment:
Both regex and glob support list of patterns (e.g. glob: `{dir1,dir2}` )  So we don't really need list here. I would change it to `Exclude string`